### PR TITLE
MINOR: Fix the flaky TBRLMM `testInternalTopicExists` test

### PR DESCRIPTION
**Pull request by kamalcph**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by clolov — COMMENTED** [review]
Sorry, if this is an internal topic why do you need to explicitly initialise it? Could you give me a bit of context on the proposed solution?

**Comment by kamalcph** [comment]
> Sorry, if this is an internal topic why do you need to explicitly initialise it? Could you give me a bit of context on the proposed solution?

The internal topic creation is asynchronous so the test gets flaky. To fix the test flakiness and in this test I want to assert that `doesTopicExist` should return true when a topic exists, so created a dummy internal topic.

File: storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java [hunk 1]
```diff
@@ -70,18 +70,20 @@ public TopicBasedRemoteLogMetadataManager topicBasedRlmm() {
...
 
     @Test
-    public void testInternalTopicExists() {
+    public void testDoesTopicExist() {
         Properties adminConfig = remoteLogMetadataManagerHarness.adminClientConfig();
         ListenerName listenerName = remoteLogMetadataManagerHarness.listenerName();
         try (Admin admin = remoteLogMetadataManagerHarness.createAdminClient(listenerName, adminConfig)) {
-            String topic = topicBasedRlmm().config().remoteLogMetadataTopicName();
+            String topic = "test-topic-exist";
+            remoteLogMetadataManagerHarness.createTopic(topic, 1, 1, new Properties(),
```
Comment (junrao) [inline storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java:78]: `createTopic` only waits for the topic metadata to be committed in the metadata log, but not waiting for the metadata to be propagated to every broker. `topicBasedRlmm().doesTopicExist` calls `describeTopic` and only checks the metadata in a least loaded broker. So, it seems that there is no strong guarantee that `doesTopicExist` is always true?

**Review by junrao — COMMENTED** [review]
@kamalcph : Thanks for the PR. Left a comment.

File: storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java (hunk 1 above)
Comment (kamalcph) [reply storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java:78]: [TestUtils#createTopicWithAdmin](https://sourcegraph.com/github.com/apache/kafka@fade3d10ea07eea5d6076b8fb1b68e2db5ffec48/-/blob/core/src/test/scala/unit/kafka/utils/TestUtils.scala?L497) waits for the metadata to be propagated to all the brokers before returning the call so the test will pass. Also, tried running the test repeatedly 100 times to ensure that the test is not flaky.

Comment (showuon) [reply storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java:78]: Yes, you're right, we'll make sure the metadata is propagated to all nodes for both KRaft and ZK modes.

**Review by showuon — APPROVED** [review]
LGTM! Thanks for the fix.

**Review by satishd — APPROVED** [review]
Thanks @kamalcph for the PR. LGTM.

File: storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java (hunk 1 above)
Comment (junrao) [reply storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java:78]: Thanks for the explanation, @kamalcph. The PR LGTM then.

**Pull request merged**